app/src/main/java/com/douglasbruce/tasky/core/domain/validation/NameValidator.kt

```diff
@@ -0,0 +1,19 @@
+package com.douglasbruce.tasky.core.domain.validation
+
+import javax.inject.Inject
+
+class NameValidator @Inject constructor() {
+
+    operator fun invoke(name: String): ValidationResult {
+        if (name.length in 4..50) {
+            return ValidationResult(
+                successful = true
+            )
+        }
+
+        return ValidationResult(
+            successful = false,
+            errorType = ErrorType.LENGTH
+        )
```

Comment:
I'm personally more a friend of using early returns for the error cases, so you can quickly scan and extend these, but I guess that's a matter of taste 😄

Reply:
Does make more sense for extendibility. Think I originally didn't want to inverse the if statement